dask/dataframe/utils.py

```diff
@@ -434,7 +434,7 @@ def check_matching_columns(meta, actual):
         if extra or missing:
             extra_info = f"  Extra:   {extra}\n  Missing: {missing}"
         else:
-            extra_info = "Order of columns does not match"
+            extra_info = f"Order of columns does not match. The expected columns order is: {actual.columns}"
```

Comment:
Adding the expected columns is great, but let's also include the metadata columns so users can see the difference in order. Otherwise there still may be questions about what the mismatch actually is. 

```suggestion
            extra_info = f"Order of columns does not match. The expected columns order is: {actual.columns}"
```

Reply:
![image](https://github.com/user-attachments/assets/e5243250-0538-431e-865c-c2fc2f104009)